### PR TITLE
Add glpk note

### PR DESCRIPTION
**Pull request by hyweldd**
This is a quick note to mention that the glpk package is required for igraph to work, and to reference the relevant igraph issue.

### CODE REVIEW

**Comment by szhorvat** [comment]
Please do not reference that issue on GitHub. We no longer post updated there. I suggest you reference the installation troubleshooting document instead: https://r.igraph.org/articles/installation-troubleshooting.html

**Comment by hyweldd** [comment]
Thanks for the feedback. The fix that worked for me on MacOS was to install glpk via homebrew. This is described in one of the comments on the issue I referenced but does not appear to be mentioned on the installation troubleshooting page.

**Comment by szhorvat** [comment]
It is not mentioned there because it won't work for most people, and because it is unnecessary unless you are explicitly choosing to compile igraph from sources. The majority of people will use the R distribution from CRAN, which means that they have access to pre-compiled binaries. How to get these is described in the troubleshooting document.

I would not encourage people to compile igraph from sources on macOS or Windows, as most of our users are not technically adept, and trying to do this will only cause difficulties without providing any benefit. The usual confusion comes from `install.packages()` asking people if they want to compile from sources. Simply answering "no" will download the last version for which a binary is available. Binary versions usually become available within a few days of a release.

Simply installing the dependencies with Homebrew on a Mac is not sufficient to compile R packages. You also need to tell R where to look for headers and libraries, usually by writing a custom `Makevars` file, which does require some expertise to do ...

Another, less common reason why people compile from sources is that they don't use CRAN's official R distribution. This means that they have no access to CRAN binaries at all, and need to compile all R packages (that use C extensions) from sources. The best advice to people who are not comfortable with doing this, and resolving compilation problems on their own, is to just use the CRAN distribution ... 

Feedback about all this, and about improving the troubleshooting document, is welcome.

**Comment by szhorvat** [comment]
> The fix that worked for me on MacOS was to install glpk via homebrew.

I'm wondering if you also installed R using Homebrew? If so, this would indeed work. If not, did you modify your Makevars file?

**Comment by szhorvat** [comment]
I hope I managed to explain the reasoning behind the current state of the Installation FAQ. The main point is: we're happy to take suggestions and make improvements, just let us know what you think is missing :-) We're also working towards making GLPK and libxml2 optional again.

**Comment by hyweldd** [comment]
I did install RStudio using homebrew, so that could well be why that solution worked for me, and possibly why renv is installing a lot packages from source.  I suspect that among Mac users instances of R installed using homebrew could be quite common, so it might be worth mentioning that for users who are running R / RStudio instances that were installed using homebrew, 'brew install glpk' is an easy potential solution to the compilation issues.

File: setup.md [hunk 1]
```diff
@@ -51,4 +51,6 @@ install.packages('igraph', repos=c(igraph = 'https://igraph.r-universe.dev',
 BiocManager::install('bluster')
 ```
 
+If this does not work, it may be necessary to check that the **glpk** package, on which igraph depends, is installed on your system. More details can be found at [https://github.com/igraph/rigraph/issues/490](https://github.com/igraph/rigraph/issues/490).
```
[the remark this answers is not in the document]
Comment (hyweldd) [reply setup.md:54]: The problem with this wording is that installation of glpk on macOS (where homebrew is often used) isn't currently described on the igraph installation troubleshooting page.

[the remark this answers is not in the document]
Comment (alanocallaghan) [reply setup.md:54]: On the main igraph page linked, it explains:

> When compiling from sources, make sure that you have C, C++ and Fortran compilers, as well as development packages for glpk and libxml2. On Debian/Ubuntu, use apt install libglpk-dev libxml2-dev. On Fedora, use yum install glpk-devel libxml2-devel.

On the troubleshooting page, it explains

> Make sure you have the GLPK C library installed

[the remark this answers is not in the document]
Comment (alanocallaghan) [reply setup.md:54]: Indeed, the text we supply either way does not reference homebrew, and given
- an igraph dev specifically asked us to not link that issue
- the brew install comment is at the bottom of the page (and could be hidden or deleted)
I think it's best to link to the official installation instructions

**Pull request merged**